app/lib/submission_builder/ty2024/states/md/documents/md502b.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module SubmissionBuilder
+  module Ty2024
+    module States
+      module Md
+        module Documents
+          class Md502b < SubmissionBuilder::Document
+            include SubmissionBuilder::FormattingMethods
+
+            def document
+              build_xml_doc("Form502B", documentId: "Form502B") do |xml|
+                xml.Dependents do
+                  xml.CountRegular calculated_fields.fetch(:MD502B_LINE_1)
+                  xml.CountOver65 calculated_fields.fetch(:MD502B_LINE_2)
+                  xml.Count calculated_fields.fetch(:MD502B_LINE_3)
+
+                  intake.dependents.each do |dependent|
+                    xml.Dependent do
+                      xml.Name do
+                        xml.FirstName sanitize_for_xml(dependent.first_name, 16)
+                        xml.MiddleInitial sanitize_for_xml(dependent.middle_initial, 1) if dependent.middle_initial.present?
+                        xml.LastName sanitize_for_xml(dependent.last_name, 32)
+                      end
+                      xml.SSN dependent.ssn
+                      xml.RelationToTaxpayer relationship(dependent)
+                      xml.ClaimedAsDependent "X"
+                      xml.Over65 "X" if dependent.senior?
+                      xml.DependentDOB dependent.dob.strftime("%Y-%m-%d")
+                    end
+                  end
+                end
+              end
+            end
+
+            private
+
+            def intake
+              @submission.data_source
+            end
+
+            def calculated_fields
+              @calculated_fields ||= intake.tax_calculator.calculate
+            end
+
+            # from MDIndividualeFileTypes.xsd
+            RELATIONSHIP_OPTIONS = {
+              "DAUGHTER": "CH",
+              "STEPCHILD": "CH",
+              "FOSTER CHILD": "FC",
+              "GRANDCHILD": "GC",
+              "SISTER": "SR",
+              "HALF SISTER": "CH",
+              "NEPHEW": "NP",
+              "STEPBROTHER": "CH",
+              "PARENT": "PT",
+              "GRANDPARENT": "GP",
+              "NONE": "NN",
+            }.freeze
+
```

Comment:
if we are using the relationship from the json in the future, would they be the non-gendered ones? like the following biologicalChild
adoptedChild
stepChild
fosterChild
grandChildOrOtherDescendantOfChild
sibling
childOfSibling

Reply:
peeking in here to say I this can stay as-is for now and I will add to the story I'm writing for updating relationship labels to include this update

Reply:
oh yeah i should have left a comment: i was told that the updated mapping was coming tomorrow at the earliest

Reply:
here's the chore for reference. it's blocked probably until sometime next week at this point